update/v1alpha1/types_clusterversion_progress_insight.go

```diff
@@ -0,0 +1,263 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterVersionProgressInsight provides summary information about an ongoing cluster control plane update in Standalone clusters
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=clusterversionprogressinsights,scope=Cluster
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2012
+// +openshift:file-pattern=cvoRunLevel=0000_00,operatorName=cluster-version-operator,operatorOrdering=02
+// +openshift:enable:FeatureGate=UpgradeStatus
```

Comment:
`UpgradeStatus` is a TechPreview featuregate. At this point I think we'd need to have a new, `DevPreview` featuregate if this is accepted.

Reply:
xref: https://github.com/openshift/api/pull/2246